app/controllers/forgot_passwords_controller.rb

```diff
@@ -0,0 +1,17 @@
+class ForgotPasswordsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
+    if user
+      user.update_attribute(:token, SecureRandom.urlsafe_base64)
```

Comment:
This part `.update_attribute(:token, SecureRandom.urlsafe_base64)` should be refactored to the model, and named properly like 'generate_token!`